### PR TITLE
lychee: 0.13.0 -> 0.14.2

### LINKED ISSUE #285614: Update request: lychee 0.13. → NEWVERSION
- Package name:
- Latest released version:

- Current version on the unstable channel:
- Current version on the stable/release channel:

- [ ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by sefidel**
## Description of changes

- Bump version `0.13.0 -> 0.14.2`

  <details>
  <summary>Changelog</summary>

  https://github.com/lycheeverse/lychee/releases/tag/v0.14.0
  https://github.com/lycheeverse/lychee/releases/tag/v0.14.1
  https://github.com/lycheeverse/lychee/releases/tag/v0.14.2
  </details>

- Remove `outputHashes` as `lychee` switched to crates.io version of `criterion`

Fixes #285614 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/lychee/default.nix [hunk 1]
```diff
@@ -9,20 +9,18 @@
...
-    hash = "sha256-JUyoOtlypDWK6HxsonVzbfQAdcXk728a8gVI/5GI2fs=";
+    hash = "sha256-6ePL76qoRDJvicMF8Hp5SDLDIyYJfgDsZyK47/DmC6U=";
   };
 
   cargoLock = {
     lockFile = ./Cargo.lock;
-    outputHashes = {
-      "criterion-0.4.0" = "sha256-0EKLRdxbH2czkZjmuaYLzkTBU687y6Iw9yqNV2TbsDw=";
-    };
+    # }
```
Comment (kirillrdy) [inline pkgs/tools/networking/lychee/default.nix:23]: it should now be possible to remove Cargo.lock and use `cargoHash` instead

Comment (sefidel) [reply pkgs/tools/networking/lychee/default.nix]: Thanks, resolved in `b2cfcce8f66a2339a64e6b7e6a1eb6fb12c1c7fa`!

**Comment by sefidel** [comment]
Result of `nixpkgs-review pr 286450` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lychee</li>
  </ul>
</details>

**Review by tuxinaut — APPROVED** [review]

**Review by totoroot — APPROVED** [review]

**Comment by kirillrdy** [comment]
x86_64-darwin build is broken, can someone have a look ?

**Comment by sefidel** [comment]
I've added `SystemConfiguration` to `buildInputs`, waiting for OfBorg to build the package.

**Comment by kirillrdy** [comment]
@sefidel  thank you !

**Pull request merged**